src/models/state/tx_creation_artifacts.rs

```diff
@@ -0,0 +1,16 @@
+use crate::models::blockchain::transaction::Transaction;
+use crate::models::state::transaction_details::TransactionDetails;
+use crate::models::state::wallet::transaction_output::TxOutput;
+
+#[derive(Debug, Clone)]
+pub(crate) struct TxCreationArtifacts {
+    pub(crate) transaction: Transaction,
+    pub(crate) details: Option<TransactionDetails>,
```

Comment:
Maybe add a (dev-)comment explaining why this isn't always present? For example, do the creation artifacts go over the wire and this saves bandwidth?